examples/linux/tcp_sock.py

```diff
@@ -0,0 +1,110 @@
+"""List all TCP sockets and their cgroup v2 paths"""
+
+import ipaddress
+import os
+import socket
+import struct
+
+from drgn import Object, cast, container_of
+from drgn.helpers.linux import (
+    TcpStates,
+    hlist_for_each,
+    hlist_nulls_empty,
+    sk_fullsock,
+    sk_nulls_for_each,
+)
+
+
+def inet_sk(sk):
+    return cast('struct inet_sock *', sk)
+
+    
+def _ipv4(be32):
+    return ipaddress.IPv4Address(struct.pack('I', be32.value_()))
+
+
+def _ipv6(in6_addr):
+    return ipaddress.IPv6Address(struct.pack('IIII', *in6_addr.in6_u.u6_addr32))
+
+
+def _brackets(ip):
+    if ip.version == 4:
+        return '{}'.format(ip.compressed)
+    elif ip.version == 6:
+        return '[{}]'.format(ip.compressed)
+    return ''
+
+
+def _ip_port(ip, port):
+    return '{:>40}:{:<6}'.format(_brackets(ip), port)
+
+
+def _cgroup_path(cgroup):
```

Comment:
This seems like it could be a useful helper.

Reply:
Yep, I have a plan to add a bunch of cgroup related helpers, but wanted to do it in a separate PR when I have cycles + not to make this one too big. LMK if it doesn't work for some reason.